### ISSUE TITLE
SPI Library Compatibility

### ISSUE THREAD

**Issue by pablosun** [MediaTek-Labs/Arduino-Add-On-for-LinkIt-SDK#78]
When using this library: https://github.com/ZinggJM/GxEPD
The SPI (Software) library does not act normally.

Issues including:
 * Does not support multiple `beginTransaction()` `endTransaction()` pairs
 * Does not allow frequency adjustment
 * Does not define `#define SPI_HAS_TRANSACTION `

**Comment by pablosun** [comment]
A temp patch for GxIO library from Will Huang:

[SPI.zip](https://github.com/MediaTek-Labs/Arduino-Add-On-for-LinkIt-SDK/files/1590652/SPI.zip)

**Comment by geoffreyvd** [comment]
yep had the same issue with the beacon example code https://github.com/twilio/linkit-one-sensor-samples/blob/master/Libraries/Adafruit_BluefruitLE_nRF51/examples/beacon/beacon.ino
